### PR TITLE
Add app permissions and attachment api - To Do API

### PR DESCRIPTION
**Pull request by avijityadav**

### CODE REVIEW

**Comment by Danielabom** [comment]
Contacted author to get info.

File: concepts/permissions-reference.md [hunk 1]
```diff
@@ -2140,7 +2140,10 @@ For more complex scenarios involving multiple permissions, see [Permission scena
 
 #### Application permissions
 
-None.
+|   Permission    |  Display String   |  Description | Admin Consent Required | Microsoft Account supported |
+|:----------------|:------------------|:-------------|:-----------------------|:--------------|
+| _Tasks.Read.All_        | Read all users’ tasks and tasklist | Allows the app to read all users’ tasks and task lists in your organization, without a signed-in user | Yes  | No |
```
Comment (Danielabom) [inline concepts/permissions-reference.md:2145]:
```suggestion
| _Tasks.Read.All_        | Read all users’ tasks and tasklist | Allows the app to read all users’ tasks and task lists in your organization without a signed-in user | Yes  | No |
```

File: concepts/permissions-reference.md [hunk 2]
```diff
@@ -2140,7 +2140,10 @@ For more complex scenarios involving multiple permissions, see [Permission scena
 
 #### Application permissions
 
-None.
+|   Permission    |  Display String   |  Description | Admin Consent Required | Microsoft Account supported |
+|:----------------|:------------------|:-------------|:-----------------------|:--------------|
+| _Tasks.Read.All_        | Read all users’ tasks and tasklist | Allows the app to read all users’ tasks and task lists in your organization without a signed-in user | Yes  | No |
+| _Tasks.ReadWrite.All_   | Read and write all users’ tasks and tasklists | Allows the app to create, read, update and delete all users’ tasks and task lists in your organization, without a signed-in user | Yes  | No |
```
Comment (Danielabom) [inline concepts/permissions-reference.md:2146]:
```suggestion
| _Tasks.ReadWrite.All_   | Read and write all users’ tasks and tasklists | Allows the app to create, read, update and delete all users’ tasks and task lists in your organization without a signed-in user | Yes  | No |
```

File: concepts/permissions-reference.md [hunk 3]
```diff
@@ -2140,7 +2140,10 @@ For more complex scenarios involving multiple permissions, see [Permission scena
 
 #### Application permissions
 
-None.
+|   Permission    |  Display String   |  Description | Admin Consent Required | Microsoft Account supported |
+|:----------------|:------------------|:-------------|:-----------------------|:--------------|
+| _Tasks.Read.All_        | Read all users’ tasks and tasklist | Allows the app to read all users’ tasks and task lists in your organization without a signed-in user | Yes  | No |
```
Comment (Danielabom) [inline concepts/permissions-reference.md:2145]:
```suggestion
| _Tasks.Read.All_        | Read all users’ tasks and tasklists| Allows the app to read all users’ tasks and task lists in your organization without a signed-in user. | Yes  | No |
```

File: concepts/permissions-reference.md [hunk 4]
```diff
@@ -2140,7 +2140,10 @@ For more complex scenarios involving multiple permissions, see [Permission scena
 
 #### Application permissions
 
-None.
+|   Permission    |  Display String   |  Description | Admin Consent Required | Microsoft Account supported |
+|:----------------|:------------------|:-------------|:-----------------------|:--------------|
+| _Tasks.Read.All_        | Read all users’ tasks and tasklists| Allows the app to read all users’ tasks and task lists in your organization without a signed-in user. | Yes  | No |
+| _Tasks.ReadWrite.All_   | Read and write all users’ tasks and tasklists | Allows the app to create, read, update and delete all users’ tasks and task lists in your organization without a signed-in user | Yes  | No |
```
Comment (Danielabom) [inline concepts/permissions-reference.md:2146]:
```suggestion
| _Tasks.ReadWrite.All_   | Read and write all users’ tasks and tasklists | Allows the app to create, read, update and delete all users’ tasks and task lists in your organization without a signed-in user. | Yes  | No |
```

**Review by Danielabom — CHANGES_REQUESTED** [review]
Dear @avijityadav , thank you for working on this PR. 
I have some questions/comments, please clarify. Thanks! 

1. I see differences between CSDL files and what we have on the api.md files. for example: I could not find any info about the taskFileAttachment resource type on the api.md file. but I see the new entity on the CSDL.

2. I believe we are missing two relationships on the api-reference/v1.0/resources/todotask.md: 

<NavigationProperty Name="attachments" Type="Collection(graph.attachmentBase)" ContainsTarget="true"/>
<NavigationProperty Name="attachmentSessions" Type="Collection(graph.attachmentSession)" ContainsTarget="true"/>

3. I do not see the changelog in this PR and I need it to properly complete my review.

File: api-reference/v1.0/resources/todotask.md [hunk 5]
```diff
@@ -47,6 +47,7 @@ This resource supports the following:
 |lastModifiedDateTime|DateTimeOffset|The date and time when the task was last modified. By default, it is in UTC. You can provide a custom time zone in the request header. The property value uses ISO 8601 format and is always in UTC time. For example, midnight UTC on Jan 1, 2020 would look like this: '2020-01-01T00:00:00Z'.|
 |recurrence|[patternedRecurrence](../resources/patternedrecurrence.md)|The recurrence pattern for the task.|
 |reminderDateTime|[dateTimeTimeZone](../resources/datetimetimezone.md)|The date and time in the specified time zone for a reminder alert of the task to occur.|
+|startDateTime|[dateTimeTimeZone](../resources/datetimetimezone.md)|The date and time in the specified time zone at which the task is scheduled to start.|
```
Comment (Danielabom) [inline api-reference/v1.0/resources/todotask.md:50]: I believe we are missing the    <Property Name="hasAttachments" Type="Edm.Boolean" /> property here.

**Comment by Danielabom** [comment]
Checking on the status of this PR. Hi @avijityadav , let me know if you have plans to make any changes this week.

File: api-reference/v1.0/resources/todotask.md (hunk 5 above)
Comment (avijityadav) [reply api-reference/v1.0/resources/todotask.md:51]: Hi @Danielabom, Added it in the latest commit.

**Comment by avijityadav** [comment]
Hi @Danielabom , 

Thanks for reviewing the PR. Answering to your questions.

**I see differences between CSDL files and what we have on the api.md files. for example: I could not find any info about the taskFileAttachment resource type on the api.md file. but I see the new entity on the CSDL.**

Avijit - There were some renaming  of few resources that API council suggested after the review was completed. Hence, please consider CSDL names as final that went in the code. 

**I believe we are missing two relationships on the api-reference/v1.0/resources/todotask.md: attachments 
  attachmentSessions**

Avijit - Added the attachments relationship. I have not added attachmentSessions here since its not a developer facing resource  and its is used internally to upload an attachment. 

**I do not see the changelog in this PR and I need it to properly complete my review.**

Avijit - Updated the changelog.

File: changelog/Microsoft.Todo.json [hunk 6]
```diff
@@ -1,5 +1,95 @@
...
+          "ChangeType": "Addition",
+          "Description": "Added the [taskFileAttachment](https://docs.microsoft.com/en-us/graph/api/resources/taskFileAttachment?view=graph-rest-1.0) resource",
+          "Target": "taskFileAttachment"
+        },
+        {
+          "Id": "8261d380-017f-4e23-bc50-34c5b12e5f3a",
+          "ApiChange": "Property",
+          "ChangedApiName": "hasAttachments",
+          "ChangeType": "Addition",
+          "Description": "Added the `hasAttachments` property to the [todoTask](https://docs.microsoft.com/en-us/graph/api/resources/todoTask?view=graph-rest-1.0) resource.",
```
Comment (Danielabom) [inline changelog/Microsoft.Todo.json:42]:
```suggestion
          "Description": "Added the **hasAttachments** property to the [todoTask](https://docs.microsoft.com/en-us/graph/api/resources/todoTask?view=graph-rest-1.0) resource.",
```

File: changelog/Microsoft.Todo.json [hunk 7]
```diff
@@ -1,5 +1,95 @@
...
+          "ChangeType": "Addition",
+          "Description": "Added the **hasAttachments** property to the [todoTask](https://docs.microsoft.com/en-us/graph/api/resources/todoTask?view=graph-rest-1.0) resource.",
+          "Target": "todoTask"
+        },
+        {
+          "Id": "8261d380-017f-4e23-bc50-34c5b12e5f3a",
+          "ApiChange": "Property",
+          "ChangedApiName": "startDateTime",
+          "ChangeType": "Addition",
+          "Description": "Added the `startDateTime` property to the [todoTask](https://docs.microsoft.com/en-us/graph/api/resources/todoTask?view=graph-rest-1.0) resource.",
```
Comment (Danielabom) [inline changelog/Microsoft.Todo.json:50]:
```suggestion
          "Description": "Added the **startDateTime** property to the [todoTask](https://docs.microsoft.com/en-us/graph/api/resources/todoTask?view=graph-rest-1.0) resource.",
```

File: changelog/Microsoft.Todo.json [hunk 8]
```diff
@@ -1,5 +1,95 @@
...
+          "Id": "8261d380-017f-4e23-bc50-34c5b12e5f3a",
+          "ApiChange": "Property",
+          "ChangedApiName": "startDateTime",
+          "ChangeType": "Addition",
+          "Description": "Added the **startDateTime** property to the [todoTask](https://docs.microsoft.com/en-us/graph/api/resources/todoTask?view=graph-rest-1.0) resource.",
+          "Target": "todoTask"
+        },
+        {
+          "Id": "8261d380-017f-4e23-bc50-34c5b12e5f3a",
+          "ApiChange": "Resource",
```
Comment (Danielabom) [inline changelog/Microsoft.Todo.json:55]:
```suggestion
          "ApiChange": "Relationship",
```

File: changelog/Microsoft.Todo.json [hunk 9]
```diff
@@ -1,5 +1,95 @@
...
+          "ChangeType": "Addition",
+          "Description": "Added the **startDateTime** property to the [todoTask](https://docs.microsoft.com/en-us/graph/api/resources/todoTask?view=graph-rest-1.0) resource.",
+          "Target": "todoTask"
+        },
+        {
+          "Id": "8261d380-017f-4e23-bc50-34c5b12e5f3a",
+          "ApiChange": "Relationship",
+          "ChangedApiName": "attachments",
+          "ChangeType": "Addition",
+          "Description": "Added the **attachments** relationship to the [todoTask](https://docs.microsoft.com/en-us/graph/api/resources/todoTask?view=graph-rest-1.0) resource.",
```
Comment (Danielabom) [inline changelog/Microsoft.Todo.json:58]:
```suggestion
          "Description": "Added the **attachments** navigation property to the [todoTask](https://docs.microsoft.com/en-us/graph/api/resources/todoTask?view=graph-rest-1.0) resource.",
```

File: changelog/Microsoft.Todo.json [hunk 10]
```diff
@@ -1,5 +1,95 @@
...
+          "Description": "Added the **attachments** navigation property to the [todoTask](https://docs.microsoft.com/en-us/graph/api/resources/todoTask?view=graph-rest-1.0) resource.",
+          "Target": "todoTask"
+        },
+        {
+          "Id": "8261d380-017f-4e23-bc50-34c5b12e5f3a",
+          "ApiChange": "Property",
+          "ChangedApiName": "startDateTime",
+          "ChangeType": "Addition",
+          "Description": "Added the **startDateTime** property to the [todoTask](https://learn.microsoft.com/en-us/graph/api/resources/todoTask?view=graph-rest-1.0) resource.",
+          "Target": "todoTask"
```
Comment (Danielabom) [inline changelog/Microsoft.Todo.json:67]: Is this a duplicated entry? Please see lines 46 to 51.

File: changelog/Microsoft.Todo.json [hunk 11]
```diff
@@ -1,5 +1,95 @@
...
+        },
+        {
+          "Id": "8261d380-017f-4e23-bc50-34c5b12e5f3a",
+          "ApiChange": "Permission",
+          "ChangedApiName": "Tasks.Read.All",
+          "ChangeType": "Addition",
+          "Description": "Added support for the application permission of `Tasks.Read.All` to the read operations of the [todoTasklist](https://learn.microsoft.com/en-us/graph/api/resources/todotasklist?view=graph-rest-v1.0) and [todoTask](https://learn.microsoft.com/en-us/graph/api/resources/todotask?view=graph-rest-v1.0) resource.",
```
Comment (Danielabom) [inline changelog/Microsoft.Todo.json:74]: Please review the description. I believe that it should be "Added support for the delegated permission."

**Review by Danielabom — CHANGES_REQUESTED** [review]
@avijityadav, thank you for working on this PR. I just have a couple of questions; please review them. Thanks!

File: changelog/Microsoft.Todo.json (hunk 11 above)
Comment (avijityadav) [reply changelog/Microsoft.Todo.json:74]: @Danielabom , We are adding only application permission. Delegated permission were included when To Do API was introduced in the year 2020.

File: changelog/Microsoft.Todo.json (hunk 10 above)
Comment (avijityadav) [reply changelog/Microsoft.Todo.json:67]: Yes, this is duplicated. I'll remove it.

**Review by Danielabom — APPROVED** [review]
@avijityadav, thank you for addressing my comments. PR approved.

File: api-reference/v1.0/api/taskfileattachment-createuploadsession.md [hunk 12]
```diff
@@ -0,0 +1,114 @@
+---
+title: "taskFileAttachment: createUploadSession"
```
Comment (Lauragra) [inline api-reference/v1.0/api/taskfileattachment-createuploadsession.md:2]: Global: The taskFileAttachment resource topic and the new methods we're adding need to be surfaced in the v1.0 TOC. Please update the TOC to include the new APIs.

File: changelog/Microsoft.Todo.json [hunk 13]
```diff
@@ -1,5 +1,87 @@
...
+          "ApiChange": "Permission",
+          "ChangedApiName": "Tasks.ReadWrite.All",
+          "ChangeType": "Addition",
+          "Description": "Added support for the `Tasks.Read.All` application permission to the read and write operations of the [todoTasklist](https://learn.microsoft.com/en-us/graph/api/resources/todotasklist?view=graph-rest-v1.0) and [todotask](https://learn.microsoft.com/en-us/graph/api/resources/todotask?view=graph-rest-v1.0) resource.",
+          "Target": "onlineMeeting"
+        }
+      ],
+      "Id": "8261d380-017f-4e23-bc50-34c5b12e5f3a",
+      "Cloud": "Prod",
+      "Version": "v1.0",
```
Comment (Lauragra) [inline changelog/Microsoft.Todo.json:80]: We don't have a changelog entry for the permission updates that we're making for beta. Please add a new ChangeList item to call out the new application permission for the beta topics.

**Comment by avijityadav** [comment]
@Lauragra , I made the edits as needed. Can you please take a look and merge if everything looks okay ?

**Review by Lauragra — APPROVED** [review]

**Pull request merged**